### ISSUE TITLE
Windows 8.1

### ISSUE THREAD

**Issue by Nukem9** [x64dbg/x64dbg#255]
Windows 8.1 Pro up to date
x64_dbg crashes the instant it's started. 

----------------------------------------
- Bitbucket: https://bitbucket.org/mrexodia/x64_dbg/issue/255
- Originally reported by: Anonymous
- Originally created at: 2015-02-20T22:41:42.195

**Comment by Nukem9** [comment]
same here, but no crash. could you try the latest snapshot http://snapshots.x64dbg.com and please include information about the crash.

----------------------------------------
Original comment by: [duncanisgoed](http://bitbucket.org/duncanisgoed)

**Comment by Nukem9** [comment]
TitanEngine is compiled with SDK for previous window versions which had different debugging structures (and code do not differentiate them basing on OS - it just uses old ones). It won't even compile on Win8.1 SDK until you fix it manually in source. So it may be unstable on new systems as guys work mainly on archaic vs2010(the worst visual version ever created)/win7 environment and possibly aren't aware of such issues.

To sum up: use it on Win8+ at your own risk ;) However it works fine in most cases, at least for me.

----------------------------------------
Original comment by: [kampber](http://bitbucket.org/kampber)

**Comment by Nukem9** [comment]
I don't really know what you're talking about. I developed x64_dbg on Windows 8.1 x64 with VS2010 (VS13 is the worst performing IDE I even encountered, especially with a few Addins installed). I never experienced any random crashes with TitanEngine. Could you point out specifically which structures you're talking about. As far as I'm aware Microsoft didn't change the DEBUG_EVENT structure since Windows 95.

I'll try compiling everything with VS2013 later this evening, but currently there is only support for compiling with VS2010.

Try the latest snapshot, it is fully tested on Win 8.1 x64.

----------------------------------------
Original comment by: [mrexodia](http://bitbucket.org/mrexodia)

**Comment by Nukem9** [comment]
Maybe I'm overreacting when talking about vs2010 but at time it was released it was not usable due to constant crashes, compatibility issues with SDKs and other things. Basically nobody used it in serious projects and simply skipped that version, but I suppose quite serious number of these could be already fixed and it is quite fine to use. If it is, then good for you.

Main points:
You can develop it on newest windows, but if you use old SDK it doesn't really matter. Even if you use the latest SDK, vs2010 can (and unfortunately - often do) quietly use headers installed with it (the old ones) over these in SDK.

I didn't experience any crashess too, I said only about compilation being based on old header files. Aand to answer the question, _FLOATING_SAVE_AREA structure no longer have Cr0NpxState member used in TitanEngine. If it compiles fine for you, then you are using old windows headers (aforementioned well-known issue with vs2010). 

Everything compiles fine on vs2013 with correct SDK headers (after little fixes on issues like this) so no need to waste your time if you don't plan to officially support it in near future. Just fix your headers :P

----------------------------------------
Original comment by: [kampber](http://bitbucket.org/kampber)

**Comment by Nukem9** [comment]
Currently VS10 is stable indeed :) Anyways, thanks for pointing out the issue with `Cr0NpxState`. It looks like they renamed it to `Spare0`:
```
typedef struct _FLOATING_SAVE_AREA {
    ULONG   ControlWord;
    ULONG   StatusWord;
    ULONG   TagWord;
    ULONG   ErrorOffset;
    ULONG   ErrorSelector;
    ULONG   DataOffset;
    ULONG   DataSelector;
    UCHAR   RegisterArea[SIZE_OF_80387_REGISTERS];
    ULONG   Spare0;
} FLOATING_SAVE_AREA;
```

I'll just remove the references to `FLOATING_SAVE_AREA.Cr0NpxState` we don't use the value anyway.

----------------------------------------
Original comment by: [mrexodia](http://bitbucket.org/mrexodia)

**Comment by Nukem9** [comment]
Great!

----------------------------------------
Original comment by: [kampber](http://bitbucket.org/kampber)

**Comment by Nukem9** [comment]
issue was resolved

----------------------------------------
Original comment by: [mrexodia](http://bitbucket.org/mrexodia)
